### PR TITLE
Aalors/add noop resolver

### PR DESCRIPTION
**Pull request by bananacocodrilo**
Added noop resolver.

This is a explicit noop resolver intended to be used by clients with their own custom balancer logic.

It expects the target to be empty as it will never resolve any target.

### CODE REVIEW

File: pkg/noopresolver/noopresolver.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	"google.golang.org/grpc/resolver"
+)
+
+// Scheme is the scheme for the noop resolver.
+const Scheme = "noop"
+
+var errInvalidTarget = errors.New("noop resolver doesn't accept a target")
+
+type noopBuilder struct{}
```
Comment (pulkit4tech) [inline pkg/noopresolver/noopresolver.go:34]: add usage for this

File: transport/tchannel/tchannel_integration_test.go [hunk 2]
```diff
@@ -49,7 +49,12 @@ import (
 func TestHandleResourceExhausted(t *testing.T) {
 	serviceName := "test-service"
 	procedureName := "test-procedure"
-	port := uint16(8000)
+
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
```
Comment (pulkit4tech) [inline transport/tchannel/tchannel_integration_test.go:53]: move this in separate PR

File: pkg/multiaddrpassthrough/multiaddrpassthrough_test.go [hunk 3]
```diff
@@ -172,7 +172,7 @@ func TestGRPCIntegration(t *testing.T) {
 
 	s := grpc.NewServer()
 	reflection.Register(s)
-	defer s.GracefulStop()
+	t.Cleanup(s.GracefulStop)
```
Comment (pulkit4tech) [inline pkg/multiaddrpassthrough/multiaddrpassthrough_test.go:175]: move this as well

**Review by pulkit4tech — APPROVED** [review]

File: pkg/noopresolver/noopresolver_test.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+			} else {
+				assert.ElementsMatch(t, cc.State.Addresses, tt.watAddress)
+				gotResolver.Close()
+			}
+		})
+	}
+}
+
+func TestClientConnectionIntegration(t *testing.T) {
+	dest := "127.0.0.1:3456"
```
Comment (pulkit4tech) [inline pkg/noopresolver/noopresolver_test.go:78]: will this cause flakyness?

Comment (bananacocodrilo) [reply pkg/noopresolver/noopresolver_test.go:78]: No, it's only using it to create a target to resolve.
Doesn't try to listen to the port or open a connection to it.
